1.19.x port

thedarkcolour opened the pull request.

thedarkcolour committed:
WIP port, compiles

thedarkcolour committed:
Move everything to deferred register. Worlds now load without immediately crashing

thedarkcolour committed:
Fix model layers and incorrect offsets in Beealyzer

thedarkcolour committed:
Fix escritoire text scaling

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/ForestryMC/ForestryMC?pullRequest=2721) <br/>All committers have signed the CLA.

thedarkcolour committed:
Almost fixed crates :)

thedarkcolour committed:
Fix two rendering bugs in alyzer and tank widget, fix pipette, fix socket, fix phantom slots

thedarkcolour committed:
Rename *Containers classes to *MenuTypes

thedarkcolour committed:
Handle packets on the main thread to avoid threading crashes using world's RandomSource

thedarkcolour committed:
Fix wood block data generation + add "wood" variants

thedarkcolour committed:
Fix sapling item model

thedarkcolour committed:
Add data generation for vanilla fireproof wood blocks

thedarkcolour committed:
Fix models for manual/managed farms

thedarkcolour committed:
Fix models for fluid containers

thedarkcolour committed:
Make crates drop items from correct position

thedarkcolour committed:
Fix broken JEI

thedarkcolour committed:
Split model data generation to improve editor performance

thedarkcolour committed:
Fixed fluid textures and added missing fabricator recipes

thedarkcolour committed:
Remove every usage of DistExecutor

thedarkcolour committed:
WIP sided tile entity ticking

thedarkcolour committed:
Sided tile entity ticking

thedarkcolour committed:
Save all changes. Currently trying to fix cocoon model, which is the only model that Minecraft still warns about.

thedarkcolour committed:
Future-proof block entity data synchronization

thedarkcolour committed:
Use Parchment for parameter mappings

thedarkcolour committed:
Fix multiblock entities not saving NBT across world loads

thedarkcolour committed:
Re-add the Peat, Clockwork, and Biogas engine + smoothing fix

thedarkcolour committed:
Increase energy storage of machines

thedarkcolour committed:
Fix mutations not showing up on client + typo in research note

thedarkcolour committed:
Fix bug where multiblocks would not load on the clientside unless in 2x2 around world origin

thedarkcolour committed:
Remove ForestryPacket.java, a useless abstraction

thedarkcolour committed:
Make IntelliJ download sources and javadocs

thedarkcolour committed:
Clean up logging

thedarkcolour committed:
Networking rewrite + remove minecart

thedarkcolour committed:
Fix jagged item rendering for Escritoire + others

thedarkcolour committed:
Code cleanup

thedarkcolour committed:
Add the Worktable

thedarkcolour committed:
Add Worktable recipe + fix model warnings

thedarkcolour committed:
Fix the rainmaker

thedarkcolour committed:
Add missing autogenerated resources

thedarkcolour committed:
Replace IToolScoop with tag

thedarkcolour committed:
Remove "Generate Data" build step

thedarkcolour committed:
Fix Apiarist and Lepidopterist backpacks

whipeoff commented:
Hello friend, sorry if I'm asking bad question or distracting you, but can you please tell, on what stage of development is mod rn and how far it is from finish?

thedarkcolour committed:
Butterfly entity rendering (AI still bugged)

thedarkcolour committed:
Butterflies can now pollinate trees

thedarkcolour committed:
Code cleanup + fix beealyzer tab

bnorax committed:
fix: fluid tank update

thedarkcolour committed:
Use null-checked level

thedarkcolour committed:
Merge branch 'bnorax-dev/1.19.x-fluid_tank_fix' into dev/1.19.x

thedarkcolour committed:
Close #3

thedarkcolour committed:
Fix natural butterfly spawning

Note: Trees with a higher sappiness and yield will increase spawn rates of butterflies. Butterfly spawns are limited within clusters, which span about a few chunks and don't limit spawns for the entire world

thedarkcolour committed:
Butterfly improvements + fix hitboxes

thedarkcolour committed:
Fix survivalist tools not giving broken forms

thedarkcolour committed:
Fix Nedelosk typo

thedarkcolour committed:
Close #8

thedarkcolour committed:
Close #2

thedarkcolour committed:
Fix destroy particles of saplings

thedarkcolour committed:
Close #6

Also fixed potential pickaxe dupe when unbreaking doesn't actually break the tool

thedarkcolour committed:
Fix fertilizer not being accepted in farm screen

thedarkcolour committed:
Fix engine crashing if placed facing up or down

thedarkcolour committed:
Add tag for valid multiblock farm bases

Also fixes nondeterministic FeatureBlockGroup & FeatureItemGroup iteration order so that cached tags don't get overwritten with unsorted contents

thedarkcolour committed:
code cleanup

thedarkcolour committed:
Close #11

thedarkcolour committed:
Fix Humus crash from unnecessary packet

thedarkcolour committed:
Fix fluid names

thedarkcolour committed:
Get rid of annoying Optionals from genetics API

thedarkcolour committed:
Leaves no longer suffocate butterflies

thedarkcolour committed:
Close #12
Fixes Analyzer block

thedarkcolour committed:
Fix escritoire display item not updating

thedarkcolour committed:
Fix escritoire sound type

thedarkcolour committed:
Better handling for NBT read failures

Note: Lots of genetics stuff reads from NBT over and over. It might be worth moving to capabilities in the future for faster access.

thedarkcolour committed:
Fix cocoons (Bombyx Mori Cocoon still unimplemented)

thedarkcolour committed:
Fix metabolism display in analyzer

thedarkcolour committed:
Fix desync behaviour in portable analyzer

Happened when inserting a specimen and then honey drops. Closing and reopening the analyzer menu would make the specimen invisible on the client.

thedarkcolour committed:
Improve bee give command + fix crash

thedarkcolour committed:
Update leaf texture

The PNG files are compressed with PNGGauntlet.

thedarkcolour committed:
Add funding links

thedarkcolour commented:
Whoops. Fork is still in progress here: https://github.com/thedarkcolour/ForestryMC
